src/com/google/enterprise/adaptor/database/UniqueKey.java

```diff
@@ -75,22 +77,22 @@
       log.fine("element: `" + e + "'");
       e = e.trim();
       String def[] = e.split(":", 2);
-      if (2 != def.length) {
-        String errmsg = "Invalid UniqueKey definition for '" + e + "'. Valid"
-            + " definition is 'column_name:type' where types can be int, string,"
-            + " timestamp, date, time and long.";
-        throw new InvalidConfigurationException(errmsg);
-      }
-      def[0] = def[0].trim();
-      def[1] = def[1].trim();
-      String name = def[0];
+      String name = def[0].trim();
       ColumnType type;
-      try {
-        type = ColumnType.valueOf(def[1].toUpperCase(US));
-      } catch (IllegalArgumentException iae) {
-        String errmsg = "Invalid UniqueKey type '" + def[1] + "'. Valid"
-            + " types are: " + ColumnType.values();
-        throw new InvalidConfigurationException(errmsg);
+
+      if (def.length == 1) {
+        // No type given. Try to get type from ResultSetMetaData later.
```

Comment:
ok.

so processing of metadata and unique key becomes more uniform. that seems good. 

addressing specific problems you've seen customers have with config is great!

i guess the remaining part is to add these checks in as simple way as possible. thank you for keeping the code as simple to follow as possible.